mern-ecommerce/controllers/user.js

```diff
@@ -40,6 +41,20 @@ exports.update = (req, res) => {
   );
 };
 
+exports.productById = (req, res, next, id) => { 
+  Product.findById(id) 
+    .populate('products.product', 'name price')
+    .exec((err, product) => { 
+      if (err || !product) {
+        return res.status(400).json({
```

Comment:
The response status code should be consistent throughout the application. In this case, it's better to define a constant for the status code at the top of the file, e.g., `const BAD_REQUEST = 400;` and use it throughout the file. This makes the code more maintainable and easier to change in the future.